### PR TITLE
feat(languages): GraphQL

### PR DESCRIPTION
**Pull request by jaredramirez**
Hey there,

This PR adds support for Graphql!

### CODE REVIEW

**Comment by the-mikedavis** [comment]
If you run `cargo xtask docgen` it'll clear up the CI :x:

**Comment by EpocSquadron** [comment]
Would it be any trouble to add this to the injections file for JavaScript, for named template strings of the form:

```js
local query = gql`...`
```

That would cover anywhere Apollo client is used, which I think is the most common library for forming queries in react and vue.

**Comment by the-mikedavis** [comment]
I think this should do it:

```lisp
; runtime/queries/javascript/injections.scm
((call_expression
   function: (identifier) @ignore
   arguments: (template_string) @injection.content)
 (#eq? @ignore "gql")
 (#set! injection.language "js"))
```

**Comment by jaredramirez** [comment]
Added!

File: runtime/queries/graphql/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+(field_definition
+  (name) @label)
+
+(object_value
+  (object_field
+    (name) @label))
+
+(enum_value
+  (name) @label)
```
Comment (archseer) [inline runtime/queries/graphql/highlights.scm:69]: Hmm, I don't think these should be `@label` but `@variable.other.member` maybe?

File: runtime/queries/javascript/injections.scm [hunk 2]
```diff
@@ -9,6 +9,14 @@
   ]
   arguments: (template_string) @injection.content)
 
+; Parse the contents of gql template literals
+
+((call_expression
+   function: (identifier) @ignore
```
Comment (pickfire) [inline runtime/queries/javascript/injections.scm:15]: Do we need to do this `@ignore`?

Comment (the-mikedavis) [reply runtime/queries/javascript/injections.scm:15]: I think something like `@_template_function_name` or something else beginning with `_` would fit well with other injections :+1:

Comment (jaredramirez) [reply runtime/queries/javascript/injections.scm:15]: I'm not sure! This is from what @the-mikedavis posted earlier in this PR: https://github.com/helix-editor/helix/pull/1515#issuecomment-1013771472

File: runtime/queries/javascript/injections.scm [hunk 3]
```diff
@@ -9,6 +9,14 @@
...
+   arguments: (template_string) @injection.content)
+ (#eq? @ignore "gql")
+ (#set! injection.language "graphql"))
```
Comment (the-mikedavis) [inline runtime/queries/javascript/injections.scm:18]:
```suggestion
((call_expression
   function: (identifier) @_template_function_name
   arguments: (template_string) @injection.content)
 (#eq? @_template_function_name "gql")
 (#set! injection.language "graphql"))
```

re: pickfire's comment. I think this reads better and looks similar to some other injections we have like rust's regex injections

`@ignore` has special meaning when it comes to tags queries so it probably doesn't make sense to use it in highlight/injection queries

**Comment by jaredramirez** [comment]
Friendly ping @archseer @the-mikedavis 😃

**Review by the-mikedavis — APPROVED** [review]
lgtm, nice work!

**Pull request merged**